File: MH/DAY6/10866.js
const path = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
let input = require("fs").readFileSync(path).toString().trim().split("\n");

const [n, ...commands] = input;
const deck = [];
const answer = [];

for (let i = 0; i < n; i++) {
  switch (commands[i]) {
    case "pop_front":
      answer.push(deck.shift() || -1);
      break;
    case "pop_back":
      answer.push(deck.pop() || -1);
      break;
    case "size":
      answer.push(deck.length);
      break;
    case "empty":
      answer.push(deck.length === 0 ? 1 : 0);
      break;
    case "front":
      answer.push(deck[0] || -1);
      break;
    case "back":
      answer.push(deck[deck.length - 1] || -1);
      break;
    default:
      if (commands[i].includes("push_front"))
        deck.unshift(commands[i].slice(11));
      else deck.push(commands[i].slice(10));
      break;
  }
}

console.log(answer.join("\n"));

File: AJM/DAY11/10866.py
import sys
from collections import deque

n = int(input())
q = deque([])
for i in range(n):
    cm = list(sys.stdin.readline().split())
    if cm[0] == "push_front":
        q.appendleft(cm[1])
    elif cm[0] == "push_back":
        q.append(cm[1])
    elif cm[0] == "pop_front":
        if len(q)!=0:
            print(q.popleft())
        else:
            print("-1")
    elif cm[0] == "pop_back":
        if len(q)!=0:
            print(q.pop())
        else:
            print("-1")
    elif cm[0] == "size":
        print(len(q))
    elif cm[0] == "empty":
        if len(q) == 0:
            print("1")
        else :
            print("0")
    elif cm[0]=="front":
        if len(q)==0:
            print("-1")
        else:
            print(q[0])
    elif cm[0]=="back":
        if len(q)==0:
            print("-1")
        else:
            print(q[-1])
        